src/main/java/edu/harvard/iq/dataverse/authorization/AuthenticationServiceBean.java

```diff
@@ -215,6 +220,7 @@ public void deleteAuthenticatedUser(Object pk) {
                 em.remove(confirmEmailData);
             }
             userNotificationService.findByUser(user.getId()).forEach(userNotificationService::delete);
+            em.createNativeQuery("delete from OAuth2TokenData where user_id =" + user.getId()).executeUpdate();
```

Comment:
can you move this to a named query?

Reply:
and actually maybe better, why not use  a cascade here too?

Reply:
I need to refresh myself on the proper setup of cascades, but sure.

Reply:
Fixed in 8afbf5e92.